extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -76,6 +97,40 @@ private function getImageAsTitleObject( $imageName ) {
 		return $title;
 	}
 
+	private function getGalleryItems( $imageData ) {
+		preg_match( '/.(UNIQ.*QINU)./U', $imageData, $galleryMarkers );
+		$galleryImages = array();
+
+		foreach ( $galleryMarkers as $marker ) {
+			$galleryWikitext = PortableInfoboxDataBag::getInstance()->getGallery( $marker );
+			$galleryLines = preg_split('/\r\n|\n|\r/', $galleryWikitext);
+
+			foreach ( $galleryLines as $galleryImage) {
```

Comment:
@SebastianMarzjan Yes. Feedback been incorporated. Btw. This is not extending gallery parsing logic - it's another yet parser for gallery.